google/generativeai/retriever.py

```diff
@@ -194,21 +193,22 @@ def list_corpora(
     if client is None:
         client = get_default_retriever_client()
 
-    request = glm.ListCorporaRequest(page_size=page_size, page_token=page_token)
-    response = client.list_corpora(request)
-    return response
+    request = glm.ListCorporaRequest(page_size=page_size)
+    for corpus in client.list_corpora(request):
+        corpus = type(corpus).to_dict(corpus)
+        yield retriever_types.Corpus(**corpus)
 
 
 async def list_corpora_async(
     *,
     page_size: Optional[int] = None,
-    page_token: Optional[str] = None,
     client: glm.RetrieverServiceClient | None = None,
-) -> list[retriever_types.Corpus]:
+) -> Iterable[retriever_types.Corpus]:
```

Comment:
```suggestion
) -> AsyncIterable[retriever_types.Corpus]:
```

Should this be `AsyncIterable`?

Reply:
Yes thanks.